### ISSUE TITLE
Extremely slow git clone on windows, full speed in linux.

### ISSUE THREAD

**Issue by rolandh** [git-for-windows/git#1283]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

Found this issue and tried the new commit. No change
https://github.com/git-for-windows/git/issues/1233#issuecomment-324158736

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
64bit

$ git --version --build-options
git version 2.14.1.windows.1.48.g37d08f901a
built from commit: 37d08f901a6a65ac562616582d7ff89d1c593b28
sizeof-long: 4
machine: x86_64

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?
Windows 10 x64 Enterprise Anniversary Update

$ cmd.exe /c ver
Microsoft Windows [Version 10.0.15063]

 - What options did you set as part of the installation? Or did you choose the
   defaults?
Tried defaults (OpenSSL Library"). Also tried using the "native windows secure channel library" with no change.

$ cat /etc/install-options.txt
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: WinSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Enabled

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other
Tried bash, cmd, powershell and windows 10 ubuntu bash

 - What commands did you run to trigger this issue? If you can provide a
$ git clone https://bitbucket.org/mirror/git.git
Cloning into 'git'...
remote: Counting objects: 225226, done.
remote: Compressing objects: 100% (58244/58244), done.
Receiving objects:   0% (1065/225226), 316.00 KiB | **190.00 KiB/s**

 - What did you expect to occur after running these commands?
8mb/s download speed

 - What actually happened instead?
 190KB/s

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Occurs on all repositories. Tried bitbucket, tried git hub and tried other public git repos.

If I run the the following on a kali linux VM (vmware) inside my windows 10 host I get the following:
$ git clone https://bitbucket.org/mirror/git.git
Cloning into 'git'...
remote: Counting objects: 225226, done.
remote: Compressing objects: 100% (58244/58244), done.
Receiving objects:   100% (225226/225226), 52.37 MiB | **8.18 MiB/s,** done.

There are more details of the problem on stackoverflow of everything else I have tried. I will duplicate here:
https://stackoverflow.com/questions/45768893/git-clone-in-windows-much-slower-than-in-linux

I have been chasing a problem with extremely slow clone speeds from github on windows. Linux (kali) does not have this problem.

I am on a 100/40 megabit/s down up connection. I can obtain close to 8mb/s download no problem.

I have plugged my PC directly into our internet outlet without any router or switches with firewall disabled with no change in speed.

Windows git clone (windows 10, git 2.14.1) consistently downloads at approxmately 150-210kb/s

I have tried another windows 7 and windows 10 machine with the same issue. I have tried a windows XP VM with the same issue. I have changed my connection to 4G cellular and speed is the same.

I have tried the github mirror test (https://bitbucket.org/mirror/git.git) with the same results.

I have tried SSH and HTTPS with a change in speed but still slow. (230kb/s vs 190-200kb/s)

I have tried other github mirrors also with slow speeds.

I have tried using a VPN (PIA) via silicon valley which was even slower.

If I use kali linux in a VM on the same machine with the network connection bridged I get ~8 mb/s from bitbucket using both SSH and HTTPS.  **I tried NAT as well also with 200kb/s indicating it is something in the windows stack causing this problem.**

If I use visual studio team services hosted build agent (windows) I get either roughly 60mb/s (possibly cached?)

I have tried running git from git bash, ubuntu bash (windows 10), windows cmd, powershell all with the same slow speeds.

I have tried the using the "OpenSSL Library" and also the "native windows secure channel library" when installing git.

I have tried downgrading as low as version 2.00 from 2.14 with no difference.

I am located in Australia.

I have spoken to bitbucket support and they have suggested that the MSS/MTU must be 1436 or lower. My router and PC defaults to 1500 however using wireshark in windows I can see that all the packets have an MTU of 1436.

Using tcpdump in kali linux I also see an MTU of 1436.

To summarise I have changed the following things: Different PC/VM Different versions of windows Different physical network connections.

I either get ~200kb/s max in windows, or I get 8mb/s max in linux.

What specifically is different between the linux and windows git implementations that could cause theses speed differences?

edit: In case git was using a slow network share (seen other questions regarding this) I tried doing a cp -r using my local clone and I get roughly 1gb/s transfer speeds (RAID 0 SSD) using the same paths as in my git config.

**Comment by dscho** [comment]
No idea whether it will help, but maybe increase `http.postbuffer` as dscribed here: https://github.com/git-for-windows/git/issues/869#issuecomment-244929786

**Comment by kgybels** [comment]
@rolandh Can you close this issue? According to [your post on stackoverflow](https://stackoverflow.com/a/46008649), this was a Windows 10 configuration issue:
>problem solved.
>
>netsh interface tcp show global
>
>netsh int tcp set global autotuninglevel=normal
>
>auto tuning was set to disabled for some reason on all my windows 10 PCs
>
>Speed instantly went to 8mb/s
